node/src/types/node_config.rs

```diff
@@ -3,11 +3,22 @@ use serde::{Deserialize, Serialize};
 
 use crate::types::BlockHash;
 
-/// Node configuration.
-#[derive(Default, DataSize, Debug, Deserialize, Serialize)]
+/// Node fast-sync configuration.
+#[derive(Default, DataSize, Debug, Deserialize, Serialize, Clone)]
 // Disallow unknown fields to ensure config files and command-line overrides contain valid keys.
 #[serde(deny_unknown_fields)]
 pub struct NodeConfig {
-    /// Hash used as a trust anchor when joining, if any.
+    /// Hash used as a trust anchor when joining. If no hash is provided, attempt to run genesis.
```

Comment:
This could be painful for operators upgrading.  It's convenient to not have to provide a trusted hash after the initial version has run.

Again, possibly not something to change in this PR, but worth speaking to SRE about at least?

Reply:
I've reverted this comment in 5f69a6797.

In addition, in 7b2f376c2 we query storage for a block header if no hash is provided.